pkg/controller/controller_utils.go

```diff
@@ -549,6 +550,11 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	}
 	prefix := getPodsPrefix(accessor.GetName())
 
+	// if it is indexed job, add index information to its generateName prefix
+	if template.Annotations != nil {
```

Comment:
This is missing the check for the existence of the annotation.

Reply:
The annotation is one part of struct`PodTemplateSpec.ObjectMeta` and it'll be defaulted as `nil` if it is not given value, why it may not exist?

Reply:
but if it doesn't exist, with this could you would get as prefix: `job-name--` (double dash)

Reply:
ah you mean the existence of the `batch.JobCompletionIndexAnnotationAlpha` in the `Annotations` map. You're right

Reply:
Ideally we should check the mode too, is it easy to pass in?

Reply:
sorry for just notice this review as the conversation resolved and hidden.
I don't know how to get the job mode information from the `runtime.Object`, otherwise we need to add additional parameter to it.
Since the `batch.JobCompletionIndexAnnotationAlpha` is only added when it is indexed job (the mode is also checked there), do we really have to add the mode check here?

Reply:
It would be so much effort to pass the Job object through that I don't think it's worth. Besides, this package is used for other controllers.

Reply:
sg